activesupport/lib/active_support/i18n_railtie.rb

```diff
@@ -55,7 +55,7 @@ def self.initialize_i18n(app)
 
       reloader = ActiveSupport::FileUpdateChecker.new(I18n.load_path.dup){ I18n.reload! }
       app.reloaders << reloader
-      ActionDispatch::Reloader.to_prepare { reloader.execute_if_updated }
+      ActionDispatch::Reloader.to_prepare { reloader.execute_if_updated.presence }
```

Comment:
Before this PR, returning `false` in an ActionDispatch callback did not have any side effect.

After this PR, returning `false` in an ActionDispatch callback halts the chain and shows a deprecation warning.

The Rails codebase uses two ActionDispatch callbacks [defined here](https://github.com/rails/rails/blob/8a1c3476a62005e3cb7cfe440da71b94fab8aa5b/actionpack/lib/action_dispatch/middleware/reloader.rb#L32L51), namely `ActionDispatch::Reloader#to_prepare` and `#to_cleanup`.

In the code above, I added `presence` to prevent the body of the `#to_prepare` callback from returning `false` and, therefore, halting the callback. In fact the `execute_if_updated` method above explicitly returns `false` sometimes ([source code](https://github.com/rails/rails/blob/8a1c3476a62005e3cb7cfe440da71b94fab8aa5b/activesupport/lib/active_support/file_update_checker.rb#L82L89)):

```ruby
def execute_if_updated
  if updated?
    execute
    true
  else
    false
  end
end
```

---

Looking back at the code now, I realize that a better alternative to the `.presence` above might be to explicitly declare a custom terminator for  `ActionDispatch::Reloader#to_prepare` and `#to_cleanup`, overriding for these specific case the default terminator, so they never halt the chain, not even when `false` is returned or `:abort` is thrown. 

In other words, replacing [these two lines](https://github.com/rails/rails/blob/master/actionpack/lib/action_dispatch/middleware/reloader.rb#L32L33):

```ruby
define_callbacks :prepare
define_callbacks :cleanup
```

with something like:

```ruby
define_callbacks :prepare, terminator: ->(target, result) { false }
define_callbacks :cleanup, terminator: ->(target, result) { false }
```

@rafaelfranca Would you rather go this direction?